### ISSUE TITLE
Query fragment error since 2.1.24

### ISSUE THREAD

**Issue by fredmontet** [datocms/gatsby-source-datocms#98]
Hi everyone!

Apparently a bug has been introduced in v2.1.24 (previous version do work) regarding the interpretation of the query fragments for pictures. During the `gatsby build` I get the following errors using Gatsby 2.19.22 (the latest as of 27.02.20):

### Output in v2.1.25
```
 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

parent is not defined

  29 |         }
  30 |         title
  31 |         titleMenu
  32 |         slug
  33 |       }
  34 |       ... on DatoCmsGallery {
  35 |         id
  36 |         model {
  37 |           apiKey
  38 |         }
> 39 |         photos {
     |         ^
  40 |           url
  41 |         }
  42 |       }
  43 |       ... on DatoCmsMap {
  44 |         id
  45 |         model {
  46 |           apiKey
  47 |         }
  48 |         latitude
  49 |         longitude
```

### Output in v2.1.24
```
success Rewriting compilation hashes - 0.006s

 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

ids is not defined

  29 |         }
  30 |         title
  31 |         titleMenu
  32 |         slug
  33 |       }
  34 |       ... on DatoCmsGallery {
  35 |         id
  36 |         model {
  37 |           apiKey
  38 |         }
> 39 |         photos {
     |         ^
  40 |           url
  41 |         }
  42 |       }
  43 |       ... on DatoCmsMap {
  44 |         id
  45 |         model {
  46 |           apiKey
  47 |         }
  48 |         latitude
  49 |         longitude
```

**Comment by stefanoverna** [comment]
@fredmontet can you try with 2.1.26?

**Comment by fredmontet** [comment]
Ouch... I get another error:

```
 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

fieldType.normalResolver is not a function

   1 | query TermsQuery {
   2 |   allDatoCmsTerm {
   3 |     edges {
   4 |       node {
   5 |         title
   6 |         slug
>  7 |         lastUpdate
     |         ^
   8 |         bodyNode {
   9 |           childMarkdownRemark {
  10 |             html
  11 |           }
  12 |         }
  13 |       }
  14 |     }
  15 |   }
  16 | }
  17 |
```

**Comment by stefanoverna** [comment]
@fredmontet please try 2.1.27!

**Comment by fbuireu** [comment]
I got exactly the same output error with another field querying some data using the `gatsby-source-datocms` plugin and I can confirm that it's now fixed with 2.1.27.

**Comment by fredmontet** [comment]
I confirm, it works!
